packages/testing-lib/src/context.ts

```diff
@@ -0,0 +1,129 @@
+export const enter: unique symbol = Symbol();
+export const exit: unique symbol = Symbol();
+
+type MaybeAsyncFunction<TArgs extends Array<any>, TReturn> =
+  | ((...args: TArgs) => TReturn)
+  | ((...args: TArgs) => PromiseLike<TReturn>);
+
+/**
+ * An object implementing the `Context` interface has some notion of
+ * "being entered" and always eventually "being exited". It should be used
+ * together with the `within` function to enclose a function with a context.
+ *
+ * A "handle" is an object that can only assumed valid while within the
+ * context. It should not be used after the context has exited and doing
+ * so should be treated as undefined behavior.
+ *
+ * The `within` function will:
+ *  - call the `enter` method exactly once.
+ *  - call the `exit` exactly once and only after `enter` has been called.
+ */
+export interface Context<THandle> {
```

Comment:
See [packages/testing-lib/src/jest.ts](https://github.com/mongodb/leafygreen-ui/pull/483/files#diff-86a0797df361701293a2a75788e063d6) for example implementation